zerver/views/user_settings.py

```diff
@@ -34,8 +35,13 @@
     EmailChangeStatus
 from confirmation.models import EmailChangeConfirmation
 
+@zulip_login_required
 def confirm_email_change(request, confirmation_key):
     # type: (HttpRequest, str) -> HttpResponse
+    user_profile = request.user
+    if user_profile.realm.email_changes_disabled:
+        raise JsonableError(_("Email change disabled"))
+
```

Comment:
Can you make this part a separate commit (can be at the start of the branch) with a test in `zerver/tests/test_email_change.py` for the new behavior (so try connecting when not logged-in and confirm you get the error)?